tests/test_E3_indicator_calculation.py

```diff
@@ -145,60 +143,140 @@ def test_total_renewable_and_non_renewable_origin_of_each_sector():
+    dict_res = {
+        PROJECT_DATA: {SECTORS: {electricity: electricity}},
+        KPI: {
+            KPI_SCALARS_DICT: {
+                TOTAL_NON_RENEWABLE_ENERGY_USE: 50,
+                TOTAL_RENEWABLE_ENERGY_USE: 50,
+            },
+            KPI_UNCOUPLED_DICT: {
+                TOTAL_RENEWABLE_ENERGY_USE: {electricity: 50},
+                TOTAL_NON_RENEWABLE_ENERGY_USE: {electricity: 50},
+            },
+        },
+    }
+
+    E3.add_renewable_factor(dict_res)
+    exp = 0.5
     assert (
-        electricity
-        in dict_renewable_energy_use[KPI][KPI_UNCOUPLED_DICT][RENEWABLE_SHARE]
-    )
+        RENEWABLE_FACTOR in dict_res[KPI][KPI_UNCOUPLED_DICT]
+    ), f"KPI {RENEWABLE_FACTOR} not added to dict_values[KPI][KPI_UNCOUPLED_DICT]"
     assert (
-        dict_renewable_energy_use[KPI][KPI_UNCOUPLED_DICT][RENEWABLE_SHARE][electricity]
-        == exp
-    )
-    assert dict_renewable_energy_use[KPI][KPI_SCALARS_DICT][RENEWABLE_SHARE] == exp
-
-
-def test_renewable_share_two_sectors():
-    # Add second sector to dict_renewable_energy_use
-    dso_h2 = "DSO_H2"
-    dict_renewable_energy_use[ENERGY_PRODUCTION].update(
-        {dso_h2 + DSO_CONSUMPTION: {ENERGY_VECTOR: h2, TOTAL_FLOW: {VALUE: flow_small}}}
-    )
-    dict_renewable_energy_use[ENERGY_PROVIDERS].update(
-        {dso_h2: {RENEWABLE_SHARE_DSO: {VALUE: 0}, ENERGY_VECTOR: h2}}
-    )
-    dict_renewable_energy_use[PROJECT_DATA][SECTORS].update({h2: h2})
+        electricity in dict_res[KPI][KPI_UNCOUPLED_DICT][RENEWABLE_FACTOR]
+    ), f"KPI {RENEWABLE_FACTOR} not defined for vector {electricity}."
+    assert (
+        dict_res[KPI][KPI_UNCOUPLED_DICT][RENEWABLE_FACTOR][electricity] == exp
+    ), f"{RENEWABLE_FACTOR} of {electricity} sector not expected value."
+    assert (
+        dict_res[KPI][KPI_SCALARS_DICT][RENEWABLE_FACTOR] == exp
+    ), f"System-wide {RENEWABLE_FACTOR} of not expected value."
+
+
+def test_renewable_factor_two_sectors():
+    dict_res = {
+        PROJECT_DATA: {SECTORS: {electricity: electricity, h2: h2}},
+        KPI: {
+            KPI_SCALARS_DICT: {
+                TOTAL_NON_RENEWABLE_ENERGY_USE: 100,
+                TOTAL_RENEWABLE_ENERGY_USE: 100,
+            },
+            KPI_UNCOUPLED_DICT: {
+                TOTAL_RENEWABLE_ENERGY_USE: {electricity: 50, h2: 50},
+                TOTAL_NON_RENEWABLE_ENERGY_USE: {electricity: 50, h2: 50},
+            },
+        },
+    }
+    E3.add_renewable_factor(dict_res)
+    assert (
+        RENEWABLE_FACTOR in dict_res[KPI][KPI_UNCOUPLED_DICT]
+    ), f"KPI {RENEWABLE_FACTOR} not added to dict_values[KPI][KPI_UNCOUPLED_DICT]"
 
-    E3.total_renewable_and_non_renewable_energy_origin(dict_renewable_energy_use)
-    E3.renewable_share(dict_renewable_energy_use)
-    assert RENEWABLE_SHARE in dict_renewable_energy_use[KPI][KPI_UNCOUPLED_DICT]
+    exp_res_sector = {
+        electricity: 0.5,
+        h2: 0.5,
+    }
+    exp = 0.5
```

Comment:
I would rename it `expected_value` as `exp` is more linked to exponential function. It could be used in the assertion failure message to mention what was the result and what is the expected value.

Reply:
018d522b1380e5e77d2077c091c04d54259ae339